packages/api-server/api_server/routes/fleets.py

```diff
@@ -1,22 +1,97 @@
+
+from ..dependencies import WithBaseQuery, base_query_params
 from ..fast_io import FastIORouter
 from ..models import Fleet, FleetState, RobotHealth
-from ..repositories import RmfRepository
+from ..models import tortoise_models as ttm
 from ..rmf_io import RmfEvents
 
 
 class FleetsRouter(FastIORouter):
-    def __init__(
-        self,
-        rmf_events: RmfEvents,
-        rmf_repo: RmfRepository,
-    ):
+    def __init__(self, rmf_events: RmfEvents, *, logger: logging.Logger):
         super().__init__(tags=["Fleets"])
 
-        @self.get("", response_model=List[Fleet])
-        async def get_fleets():
-            return await rmf_repo.query_fleets()
+        class GetFleetsResponse(Pagination.response_model(Fleet)):
+            pass
+
+        @self.get("", response_model=GetFleetsResponse)
+        async def get_fleets(
+            with_base_query: WithBaseQuery[ttm.FleetState] = Depends(
+                base_query_params({"fleet_name": "id_"})
+            ),
+            fleet_name: Optional[str] = Query(
+                None, description="comma separated list of fleet names"
+            ),
+        ):
+            filter_params = {}
+            if fleet_name is not None:
+                filter_params["id___in"] = fleet_name.split(",")
+            states = await with_base_query(ttm.FleetState.filter(**filter_params))
+            results: Pagination[Fleet] = Pagination(
+                total_count=states.total_count,
+                items=[Fleet(name=s.id_, state=s.data) for s in states.items],
+            )
+            return results
+
+        class GetRobotsResponse(Pagination.response_model(Robot)):
+            pass
+
+        @self.get("/robots", response_model=GetRobotsResponse)
+        async def get_robots(
+            with_base_query: WithBaseQuery[ttm.RobotState] = Depends(
+                base_query_params()
+            ),
+            fleet_name: Optional[str] = Query(
+                None, description="comma separated list of fleet names"
+            ),
+            robot_name: Optional[str] = Query(
+                None, description="comma separated list of robot names"
+            ),
+        ):
+            filter_params = {}
+            if fleet_name is not None:
+                filter_params["fleet_name__in"] = fleet_name.split(",")
+            if robot_name is not None:
+                filter_params["robot_name__in"] = robot_name.split(",")
+
+            robot_states = await with_base_query(ttm.RobotState.filter(**filter_params))
+            robots = {
+                f"{q.fleet_name}/{q.robot_name}": Robot(
+                    fleet=q.fleet_name, name=q.robot_name, state=q.data
+                )
+                for q in robot_states.items
+            }
+
+            filter_states = [
+                TaskStateEnum.ACTIVE.value,
+                TaskStateEnum.PENDING.value,
+                TaskStateEnum.QUEUED.value,
+            ]
```

Comment:
@vallq Is this the correct task states that should be displayed on the robot panels?

Reply:
hi KP, sorry for the delayed reply!! yes, just these 3 tasks states